Exclude Konflux application name from image name

tkdchen opened the pull request:
STONEBLD-2693

tkdchen commented:
/test image-controller-e2e

tkdchen commented:
/test image-controller-e2e

tnevrlka commented:
The reason behind this change wasn't clear to me until I've read the story.
 `Since Components have unique name, it's safe to remove application name from the image name`
However, upstream users don't have access to the `STONEBLD` Jira project. 
Could we start including more information in the commit message?

tkdchen commented:
@tnevrlka Good point. To save time from waiting the e2e-tests again, I updated the PR description this time. I also added an item for an improvement in the retro doc.
